src/gfx/pal_packing.cpp

```diff
@@ -0,0 +1,359 @@
+/*
+ * This file is part of RGBDS.
+ *
+ * Copyright (c) 2022, Eldred Habert and RGBDS contributors.
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#include "gfx/pal_packing.hpp"
+
+#include <assert.h>
+#include <bitset>
+#include <inttypes.h>
+#include <numeric>
+#include <optional>
+#include <queue>
+#include <tuple>
+#include <type_traits>
+#include <unordered_set>
+#include <vector>
+
+#include "gfx/main.hpp"
+#include "gfx/proto_palette.hpp"
+
+using std::swap;
+
+namespace packing {
+
+// The solvers here are picked from this paper: http://arxiv.org/abs/1605.00558
```

Comment:
I'd like to have the paper title and authors mentioned here too.

Reply:
Shouldn't we acknowledge them in the README?

Reply:
Sure, there too, but at least having the paper title right where it's relevant gives you something to search for beyond an opaque URL that might die.